### ISSUE TITLE
Support for __post_init__ and init=False

### ISSUE THREAD

**Issue by djbower** [ajatkj/typed_configparser#10]
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything

### Description

Initialising derived parameters in a data class is often done in __post_init__(), which is called by __init__().  In this regard, `typed-configparser` would need to do the same since it defines its own data class `__init__`, which currently does not call `__post_init__` (if it is available).

I copy-paste the below from the standard python documentation (https://docs.python.org/3/library/dataclasses.html).  It would also be nice if `typed-configparser` could ignore the `c` parameter and not try to populate it from the configuration section (since here `init=False`). Then the following example structure would be supported by `typed-configparser`, which it quite a common construction for those using data classes:

```
@dataclass
class C:
    a: float
    b: float
    c: float = field(init=False)

    def __post_init__(self):
        self.c = self.a + self.b
```

**Comment by ajatkj** [comment]
For init=False, how do you suggest it be handled? 
Scenario 1: init=False but option present in the config
Scenario 2: init=False, no option in the config and no __post_init__ as well

I think for scenario 1, option should be updated from config
for scenario 2, init it as None. It is then left to the users to handle this field.

**Comment by djbower** [comment]
Yes I agree. That sounds like sensible intuitive behaviour for both scenarios.

**Comment by ajatkj** [comment]
Changes released in version 1.1.0 (also available on pypi)
